addons/event/views/event_views.xml

```diff
@@ -54,8 +54,7 @@
                                <field name="event_type_mail_ids" class="w-100">
                                    <tree string="Communication" editable="bottom">
                                        <field name="notification_type"/>
-                                       <field name="template_model_id" invisible="1"/>
-                                       <field name="template_ref" options="{'model_field': 'template_model_id', 'no_quick_create': True}" context="{'filter_template_on_event': True, 'default_model': 'event.registration'}"/>
+                                       <field name="template_ref" options="{'hide_model': True, 'no_quick_create': True}" context="{'filter_template_on_event': True, 'default_model': 'event.registration'}"/>
```

Comment:
I think this view is about event.type.mail record, but the onchange are only added to the event.mail model

so this change is good, but you also need to add similar onchange to event.type.mail model :+1: